manim/mobject/geometry.py

```diff
@@ -1554,3 +1556,228 @@ def __init__(self, main_shape, *mobjects, **kwargs):
+                    ArcAngle(line1, line2, other_angle=True),
+                    ArcAngle(line1, line2, radius=0.4, quadrant=(1,-1)),
+                    ArcAngle(line1, line2, radius=0.5, quadrant=(-1,1), stroke_width=8),
+                    ArcAngle(line1, line2, radius=0.7, quadrant=(-1,-1), color=RED, other_angle=True),
+                ]
+                line_list = VGroup( *[VGroup() for k in range(8)] )
+                for k in range(8):
+                    linea = line1.copy()
+                    lineb = line2.copy()
+                    line_list[k].add( linea )
+                    line_list[k].add( lineb )
+                    line_list[k].add( arcangles[k] )
+                line_list.arrange_in_grid(n_rows=2, n_cols=4, buff=1.5)
+                self.add(
+                    line_list
+                )
+
+    """
+
+    def __init__(
+        self,
+        line1,
+        line2,
+        radius=0.3,
+        quadrant=(1, 1),
+        other_angle=False,
+        dot=False,
+        dot_radius=None,
+        dot_distance=0.55,
+        dot_color=WHITE,
+        **kwargs
+    ):
+        self.radius = radius
+        self.quadrant = quadrant
+        self.dot_distance = dot_distance
+        inter = line_intersection(
+            [line1.get_start(), line1.get_end()], [line2.get_start(), line2.get_end()]
+        )
+        anchor_angle_1 = inter + quadrant[0] * radius * line1.get_unit_vector()
+        anchor_angle_2 = inter + quadrant[1] * radius * line2.get_unit_vector()
+
+        angle_1 = angle_of_vector(anchor_angle_1 - inter)
+        angle_2 = angle_of_vector(anchor_angle_2 - inter)
+
+        if other_angle == False:
+            start_angle = angle_1
+            if angle_2 > angle_1:
+                angle_fin = angle_2 - angle_1
+            else:
+                angle_fin = 2 * np.pi - (angle_1 - angle_2)
+        else:
+            start_angle = angle_1
+            if angle_2 < angle_1:
+                angle_fin = -angle_1 + angle_2
+            else:
+                angle_fin = -2 * np.pi + (angle_2 - angle_1)
+
+        Arc.__init__(
+            self,
+            radius=radius,
+            angle=angle_fin,
+            start_angle=start_angle,
+            arc_center=inter,
+            **kwargs
+        )
+        if dot == True:
+            if dot_radius == None:
+                dot_radius = radius / 10
+            else:
+                self.dot_radius = dot_radius
+            right_dot = Dot(ORIGIN, radius=dot_radius, color=dot_color)
+            dot_anchor = (
+                inter
+                + (self.get_center() - inter)
+                / np.linalg.norm(self.get_center() - inter)
+                * radius
+                * dot_distance
+            )
+            right_dot.move_to(dot_anchor)
+            self.add(right_dot)
+
+
+class RightAngle(VMobject):
+    """An elbow-type mobject representing a right angle between two lines.
```

Comment:
I changed to inherit from both, `Arc` and `Elbow`, but my naive code isn't working. It has something to do with multiple inheritances, that is I get `AttributeError: Angle object has no attribute 'start_angle'`when running an `elbow=True` script. I have no experience with that and I don't know how easy of a fix that is. Any ideas?

Reply:
This has to do with the [method resolution order](http://www.srikanthtechnologies.com/blog/python/mro.aspx) in multiple inheritance... what's happening is Elbow is executing `__init__` which calls the parent classes' `__init__` up the hierarchy to Mobject. At this point in `Mobject.__init__` makes a call to [generate_points()](https://github.com/PhotonSpheres/manim/blob/b56c6949e8c8a7fd500138b90ec4fca17337b51f/manim/mobject/mobject.py#L61) and since Arc comes first in the multiple inheritance, it uses `Arc.generate_points()` which fails since Elbow doesn't have `self.start_angle` (but Arc does).

A couple of solutions in order of preference:
1. Add Elbow.generate_points() (moving most of Elbow's `__init__()` to `generate_points`), and add `Angle.generate_points` to call the correct generate_points:
```py
class Angle(Arc, Elbow):
# ....
    def generate_points(self):
        if self.elbow:
            Elbow.generate_points(self)
        else:
            Arc.generate_points(self)
```
^ Note: there's likely a better/more "pythonic" method than this. If anyone else has ideas, feel free to chime in as there's not many examples of multiple inheritance in this library.

2. Inherit from VMobject in Angle instead (not recommended). This will serve to call VMobject.generate_points when doing Elbow.__init__(), but at the cost of losing the dual inheritance.

Reply:
Thanks for the help! Unfortunately, I am not really sure how to implement your first idea. Do you have a small working example of such a script I could look at?

Reply:
Absolutely, here's a [quick commit](https://github.com/jsonvillanueva/manim/commit/7844569fc7390da5df3a543ad662f4a04b3ff879) off your latest changes. It seems you don't actually need to split the logic off of Elbow into `generate_points()`. It's enough to just override it in `Angle`.

Reply:
Also, there should be some test cases for this PR -- just in the event that something in the future changes we know if any expected behavior breaks/changes. Read up on [this section](https://github.com/ManimCommunity/manim/wiki/Testing#graphical-unit-test) on graphical unit testing which involves creating control data (.npz file) for your new Angle object. I think this will complete the PR (in my eyes at least).

Reply:
Oh, that is very neat, cheers!

Reply:
I added GU tests for `Angle` and `RightAngle`. I am not sure how complicated they should be and I only added the angles to the screen, not the lines. Would that be a good idea?

Reply:
Adding only the angles is fine since that's ultimately what's being tested, but maybe an additional test case with the lines is necessary. I do believe we could use increased complexity and really get at some of the edge cases (like the case I brought up about shorter line segments/dynamic sizing of the radius). I think your examples for the docs would also serve as a good test (a multitude of angles being tested all at the same time in one frame).

Speaking of your docs... they didn't render. Will have to look into that. Are they rendering locally for you?